cdap-cli/src/main/java/co/cask/cdap/cli/command/view/CreateOrUpdateViewCommand.java

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * Creates or updates a view.
+ */
+public class CreateOrUpdateViewCommand extends AbstractAuthCommand {
+
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .create();
+  private final ViewClient client;
+
+  @Inject
+  public CreateOrUpdateViewCommand(ViewClient client, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.client = client;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    Id.View viewId = Id.View.from(cliConfig.getCurrentNamespace(), arguments.get(ArgumentName.VIEW.toString()));
+    Id.Stream streamId = Id.Stream.from(cliConfig.getCurrentNamespace(), arguments.get(ArgumentName.STREAM.toString()));
+    String formatName = arguments.get(ArgumentName.FORMAT.toString());
+    Schema schema = getSchema(arguments);
+
+    Map<String, String> settings = Collections.emptyMap();
+    if (arguments.hasArgument(ArgumentName.SETTINGS.toString())) {
+      settings = ArgumentParser.parseMap(arguments.get(ArgumentName.SETTINGS.toString()));
+    }
+    FormatSpecification formatSpecification = new FormatSpecification(formatName, schema, settings);
+    ViewConfig viewConfig = new ViewConfig(streamId, formatSpecification);
+
+    boolean created = client.createOrUpdate(viewId, viewConfig);
+    if (created) {
+      output.printf("Successfully created view '%s'\n", viewId.getId());
+    } else {
+      output.printf("Successfully updated view '%s'\n", viewId.getId());
+    }
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("create view <%s> stream <%s> format <%s> [schema <%s>] [settings <%s>]",
+                         ArgumentName.VIEW, ArgumentName.STREAM, ArgumentName.FORMAT,
+                         ArgumentName.SCHEMA, ArgumentName.SETTINGS);
+  }
+
+  @Override
+  public String getDescription() {
+    return new StringBuilder()
+      .append("Creates or updates ")
+      .append(Fragment.of(Article.A, ElementType.VIEW.getName()))
+      .append(". Valid <").append(ArgumentName.FORMAT).append(">s are ")
+      .append(Joiner.on(", ").join(Formats.ALL))
+      .append(". <")
+      .append(ArgumentName.SCHEMA)
+      .append("> is a sql-like schema \"column_name data_type, ...\" or Avro-like JSON schema and <")
+      .append(ArgumentName.SETTINGS)
+      .append("> is specified in the format \"key1=v1 key2=v2\".")
+      .toString();
+  }
+
+  private Schema getSchema(Arguments arguments) throws IOException {
+    Schema schema = null;
+    if (arguments.hasArgument(ArgumentName.SCHEMA.toString())) {
+      // if it's a json object, try to parse it as json
+      String schemaStr = arguments.get(ArgumentName.SCHEMA.toString());
+      schema = isJsonObject(schemaStr) ? Schema.parseJson(schemaStr) : Schema.parseSQL(schemaStr);
+    }
+    return schema;
+  }
+
+  private boolean isJsonObject(String str) {
+    try {
+      GSON.fromJson(str, JsonObject.class);
```

Comment:
So you are actually parsing it twice? Can you just do `Schema.parseJson` and I believe it will throw exception if it is not json?